documentation/library_guide/introduction.rst

```diff
@@ -63,6 +63,8 @@ To use Parallel API with the device execution policies, you need to install the
 Difference with Standard C++ Parallel Algorithms
 ************************************************
 
+* Function object parameters passed in to algorithms executed with device policies must provide `const` function call
```

Comment:
I think adding something like that is useful. But probably it should be stated differently.

What we have here is not sufficient because SYCL specification also says that writing to "member variables" (I really hope that they meant "non-static data members") in the SYCL kernel is undefined behavior. In simple words it means that even if `operator()` is `const`-qualified user is not allowed to make members of the class as `mutable` and attempt to right to those.

That makes me think that we need to either reference to SYCL specification, or make a slightly different statement in our doc, or do some combination of the two mentioned approaches.

Reply:
This is a good point.  We could make it a bit stronger with something like:

"Function object parameters passed in to algorithms executed with device policies must provide `const` function call operators. The SYCL spec (4.12) states that writing to such an object during a SYCL kernel is undefined behavior." 

This should make it more clear that even marking as mutable within a wrapper is not supported, if you are in fact writing to the mutable object.

Reply:
I like the updated instance. I think that it is very clear and I have no updates to the text.

Reply:
Updated the language to what I proposed.